web-api/src/Kerbero.Domain/SmartLockKeys/Interactors/CreateSmartLockKeyInteractor.cs

```diff
@@ -1,32 +1,58 @@
 using FluentResults;
-using Kerbero.Domain.SmartLockKeys.Managers;
+using Kerbero.Domain.Common.Errors;
+using Kerbero.Domain.NukiCredentials.Repositories;
+using Kerbero.Domain.SmartLockKeys.Dtos;
+using Kerbero.Domain.SmartLockKeys.Interfaces;
 using Kerbero.Domain.SmartLockKeys.Mappers;
-using Kerbero.Domain.SmartLockKeys.Models.PresentationRequests;
-using Kerbero.Domain.SmartLockKeys.Models.PresentationResponses;
 using Kerbero.Domain.SmartLockKeys.Repositories;
+using Kerbero.Domain.SmartLockKeys.Utils;
+using Kerbero.Domain.SmartLocks.Models;
+using Kerbero.Domain.SmartLocks.Repositories;
 
 namespace Kerbero.Domain.SmartLockKeys.Interactors;
 
-public class CreateSmartLockKeyInteractor
+public class CreateSmartLockKeyInteractor: ICreateSmartLockKeyInteractor
 {
-	private readonly ISmartLockKeyGeneratorManager _smartLockKeyGeneratorManager;
-	private readonly ISmartLockKeyPersistentRepository _smartLockKeyPersistentRepository;
+	private readonly ISmartLockKeyRepository _smartLockKeyRepository;
+	private readonly INukiSmartLockRepository _nukiSmartLockRepository;
+	private readonly INukiCredentialRepository _nukiCredentialRepository;
 
-	public CreateSmartLockKeyInteractor(ISmartLockKeyGeneratorManager smartLockKeyGeneratorManager, ISmartLockKeyPersistentRepository smartLockKeyPersistentRepository)
+	public CreateSmartLockKeyInteractor(
+		ISmartLockKeyRepository smartLockKeyRepository,
+		INukiSmartLockRepository nukiSmartLockRepository,
+		INukiCredentialRepository nukiCredentialRepository)
 	{
-		_smartLockKeyGeneratorManager = smartLockKeyGeneratorManager;
-		_smartLockKeyPersistentRepository = smartLockKeyPersistentRepository;
+		_smartLockKeyRepository = smartLockKeyRepository;
+		_nukiSmartLockRepository = nukiSmartLockRepository;
+		_nukiCredentialRepository = nukiCredentialRepository;
 	}
 
-	public async Task<Result<CreateSmartLockKeyPresentationResponse>> Handle(CreateSmartLockKeyPresentationRequest presentationRequest)
+	public async Task<Result<SmartLockKeyDto>> Handle(CreateSmartLockKeyParams @params)
 	{
-		var generatedKeyResult = _smartLockKeyGeneratorManager.GenerateSmartLockKey(presentationRequest.SmartLockId, presentationRequest.ExpiryDate);
+		if (@params.SmartLockProvider != SmartLockProvider.Nuki)
+		{
+			return Result.Fail(new UnsupportedSmartLockProviderError());
+		}
+		var nukiCredentialResult = await _nukiCredentialRepository.GetById(@params.CredentialId);
+		if (nukiCredentialResult.IsFailed)
+		{
+			return Result.Fail(nukiCredentialResult.Errors);
+		}
 
-		var createResult = await _smartLockKeyPersistentRepository.Create(generatedKeyResult.Value);
-		if (createResult.IsFailed)
+		var nukiSmartLock = await _nukiSmartLockRepository.Get(nukiCredentialResult.Value, @params.SmartLockId);
+		
+		if (nukiSmartLock.IsFailed)
+		{
+			return Result.Fail(nukiCredentialResult.Errors);
+		}
+		
+		var generatedKeyResult = SmartLockKeyUtils.CreateKey(@params.SmartLockId, @params.ExpiryDate, @params.CredentialId);
```

Comment:
```suggestion
		var generatedKeyResult = SmartLockKeyModel.CreateKey(@params.SmartLockId, @params.ExpiryDate, @params.CredentialId);
```
This is a perfect example of business logic that should stay inside the domain entity of this feature